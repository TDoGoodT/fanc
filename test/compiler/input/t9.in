int increment(int a) {
    return a + 1;
}
int decrement(int a) {
    return a - 1;
}
int subtract(int a, int ab) {
    int tmpa = a;
    int tmpb = ab;
    while (tmpb > 0) {
        tmpa = decrement(tmpa);
        tmpb = decrement(tmpb);
    }
    return tmpa;
}
int divide(int a, int ab) {
    int quotient = 0;
    int tmpa = a;
    while (tmpa >= ab) {
        tmpa = subtract(tmpa, ab);
        quotient = increment(quotient);
    }
    return quotient;
}

void main() {
    int num1 = 10;
    int num2 = 2;

    int result = divide(num1, num2);

    print("Division of 10 by 2: ");
    printi(result);
}
