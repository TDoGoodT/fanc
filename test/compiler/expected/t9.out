Division of 10 by 2: 
5
